### PR TITLE
feat: add support for property and parameter maps too

### PR DESCRIPTION
**Pull request by mhmd-azeez**
Related to https://github.com/dylibso/xtp/pull/852

### CODE REVIEW

File: tests/index.test.ts [hunk 1]
```diff
@@ -97,14 +112,55 @@ const testSchema = {
...
+        "description": "A map of map schema",
+        "type": "object",
+        "additionalProperties": {
+          "type": "object",
           "additionalProperties": {
-              "type": "string"
+            "type": "string"
           }
+        }
+      },
```
Comment (bhelx) [inline tests/index.test.ts]: I'm fine with this, but do we really need to support recursive types? I'm okay with denying this pattern if it overly complicates the bindgens.

Comment (bhelx) [reply tests/index.test.ts]: I think it would be pretty rare. If you have deeply nested, typed maps, that's probably best being an object.

Comment (mhmd-azeez) [reply tests/index.test.ts]: I think I am going to do that, this should cover most of the use cases while keeping the bindgens sane

**Comment by bhelx** [comment]
Putting this feature on ice for a bit

**Comment by nilslice** [comment]
@bhelx,
> Putting this feature on ice for a bit

Are we anticipating the top-level map feature still? (like additionalProperties for a component.schema type), or is all map-related functionality being held off for the time being?

**Comment by mhmd-azeez** [comment]
Closing in favor of #18

**Pull request closed without merging**